apps/examples/messaging_sample/messaging_main.c

```diff
@@ -58,36 +58,49 @@
 #include <stdlib.h>
 #include "messaging_sample_internal.h"
 
-#ifdef CONFIG_BUILD_KERNEL
-int main(int argc, FAR char *argv[])
-#else
-int messaging_main(int argc, char *argv[])
-#endif
+static void messaging_sample_run(int loop_cnt)
 {
-	int loop_cnt;
+	printf("==Messaging Sample %d-th Iteration!==\n", loop_cnt);
 
-	printf("=== Start Messaging Sample : %d-times will run.\n", CONFIG_EXAMPLES_MESSAGING_SAMPLE_LOOPS);
-	for (loop_cnt = 0; loop_cnt < CONFIG_EXAMPLES_MESSAGING_SAMPLE_LOOPS; loop_cnt++) {
+	noreply_nonblock_messaging_sample();
 
-		printf("==Messaging Sample %d-th Iteration!==\n", loop_cnt + 1);
+	/* Wait for finishing previous test. */
+	sleep(1);
 
-		noreply_nonblock_messaging_sample();
+	sync_block_messaging_sample();
 
-		/* Wait for finishing previous test. */
-		sleep(1);
+	/* Wait for finishing previous test. */
+	sleep(1);
 
-		sync_block_messaging_sample();
+	multicast_messaging_sample();
 
-		/* Wait for finishing previous test. */
-		sleep(1);
+	/* Wait for finishing previous test. */
+	sleep(1);
 
-		multicast_messaging_sample();
+	printf("\n=== Messaging Sample is finished! (run %d-times) ===\n", loop_cnt);
+}
 
-		/* Wait for finishing previous test. */
-		sleep(1);
-	}
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int messaging_main(int argc, char *argv[])
+#endif
+{
+	int loop_cnt;
 
-	printf("\n=== Messaging Sample is finished! (run %d-times) ===\n", loop_cnt);
+	printf("=== Start Messaging Sample : %d-times will run.\n", CONFIG_EXAMPLES_MESSAGING_NUM_OF_REPETITIONS);
```

Comment:
Its  better to add this print inside if-else since CONFIG_EXAMPLES_MESSAGING_NUM_OF_REPETITIONS can be negative also.